bitcoin/src/blockdata/witness.rs

```diff
@@ -348,8 +348,8 @@ mod test {
 
     #[test]
     fn consensus_serialize() {
-        let el_0 = hex!("03d2e15674941bad4a996372cb87e1856d3652606d98562fe39c5e9e7e413f2105");
-        let el_1 = hex!("000000");
+        let el_0 = hex!("03d2e15674941bad4a996372cb87e1856d3652606d98562fe39c5e9e7e413f2105").to_vec();
+        let el_1 = hex!("000000").to_vec();
```

Comment:
These could've been just slices.

Reply:
I now see that for some reason this tests that `Vec<Vec<u8>>` serialization is the same as `Witness`. Do we actually need this? @apoelstra I think since we added `Witness` `Vec<Vec<u8>>` serialization is no longer relevant.

Reply:
> I now see that for some reason this tests that `Vec<Vec<u8>>` serialization is the same as `Witness`.

Well, because that's how the serialization of `Witness` is defined.

Maybe we can drop the encoding impls for `Vec<Vec<u8>>`, though it's a single line of code and isn't hurting anything and makes this test easy to write -- although it should probably be an arbitrary/kani test rather than a fixed-vector one.

Reply:
> Well, because that's how the serialization of `Witness` is defined.

My idea is to move the entire thing to `Witness` - indeed, removing `Vec<Vec<u8>>`.

> isn't hurting anything

If any other future soft fork adds a structure that's similar to witness but encoded differently it'll be a mess. I would even argue it already has something similar - taproot merkle branch. Though we have a type that's so much more convenient nobody will ever convert it so it's not a problem now. Note that I don't object to a function capable of serializing `Vec<Vec<u8>>` existing, just to it being the trait impl.

And yes, I think we should fuzz it.

Reply:
I don't think that taproot merkle branches are similar at all to a Vec<Vec<u8>>. They are similar to a Vec<[u8; 32]>, and without looking I can't say whether they are serialized like that or like a Vec<u8>.

But regardless, it's a distinct object which has its own distinct type, and the "raw type" that shares its serialization means little more than a mnemonic for remembering/implementing/testing the serialization.

I don't really mind removing Vec<Vec<u8>>, but I really don't think it hurts anything.